reflex/state.py

```diff
@@ -3467,20 +3458,16 @@ async def _wait_lock(self, lock_key: bytes, lock_id: bytes) -> None:
                 raise
         async with self.redis.pubsub() as pubsub:
             await pubsub.psubscribe(lock_key_channel)
-            while not state_is_locked:
-                # wait for the lock to be released
-                while True:
-                    if not await self.redis.exists(lock_key):
-                        break  # key was removed, try to get the lock again
-                    message = await pubsub.get_message(
-                        ignore_subscribe_messages=True,
-                        timeout=self.lock_expiration / 1000.0,
-                    )
-                    if message is None:
-                        continue
-                    if message["data"] in self._redis_keyspace_lock_release_events:
-                        break
-                state_is_locked = await self._try_get_lock(lock_key, lock_id)
+            # wait for the lock to be released
+            while True:
+                # fast path
+                if await self._try_get_lock(lock_key, lock_id):
+                    return
+                # wait for lock events
+                _ = await pubsub.get_message(
```

Comment:
we could add a small wrapper around get_message which filters for _redis_keyspace_lock_release_events and calls get_message again with the time left

Reply:
done in https://github.com/reflex-dev/reflex/pull/4456/commits/368a2ce55fd7286ecfbb8657c86bf821a1330ac5